### PR TITLE
[DOP-21387] Rebuild documentation structure

### PR DESCRIPTION
**Pull request by dolfinus**
## Change Summary

* Added `docs/quickstart` section with instruction how run all Data.Rentgen components in just one command, and how to connect Spark & Airflow to Kafka.
* Added `docs/entities` page describing all entities in Data.Rentgen, and their relations.
* Renamed `docs/backend` to `docs/reference`. For each section, like `docs/reference/server`, `docs/reference/consumer`, move installation instruction to `index.rst`, and rewrite it a bit.
* Added `docs/reference/database` section describing how to run PostgreSQL (with and without docker). Moved here instruction how to run migrations and create partitions (previously this was described twice, in server and in consumer docs). Added note that partition creation script should be run on schedule.
* Added `docs/reference/broker` section describing how to run Kafka.
* Added frontend container to docker-compose.yml and `docs/reference/frontend` describing how to start and configure it.

It's a bit hard to read the diff, please see https://data-rentgen--108.org.readthedocs.build/en/108/

## Checklist

* [x] Commit message and PR title is comprehensive
* [ ] Keep the change as small as possible
* [ ] Unit and integration tests for the changes exist
* [ ] Tests pass on CI and coverage does not decrease
* [x] Documentation reflects the changes where applicable
* [ ] `docs/changelog/next_release/<pull request or issue id>.<change type>.rst` file added describing change
  (see [CONTRIBUTING.rst](https://github.com/MobileTeleSystems/data-rentgen/blob/develop/CONTRIBUTING.rst) for details.)
* [x] My PR is ready to review.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -31,24 +31,26 @@
...
 -----
 
-* Collect lineage events produced by OpenLineage clients & integrations (Spark, Airflow, Flink, custom ones).
+* Collect lineage events produced by OpenLineage clients & integrations (Spark, Airflow).
+* Support consuming large amounts of lineage events, by using Kafka as event buffer and storing data in tables partitioned by event timestamp.
 * Store operation-grained events (instead of job grained `Marquez <https://marquezproject.ai/>`_), for better detalization.
-* Provide API for run ↔ dataset lineage, as well as parent run → children run lineage.
-* Support handling large amounts of lineage events, using Kafka as event buffer and storing data in tables partitioned by event timestamp.
+* Provide API for building run ↔ dataset lineage, as well as parent run → children run lineage.
+* Ability to build lineage graph with specific time boundaries (unlike Marquez there lineage is build only for last job run).
```
Comment (TiGrib) [inline README.rst:45]: Maybe move all compares to separate section?

File: docs/quickstart/setup_spark.rst [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+* Perform some data operations using Spark, like:
+
+.. code:: python
+
+    df = spark.read.format("jdbc").options(...).load()
+    df.write.format("csv").save("/output/path")
+
+
+* After ETL pipeline is finished, **stop** Spark session (this is important!):
```
Comment (TiGrib) [inline docs/quickstart/setup_spark.rst:82]: MB add:  why it is important ?

**Review by TiGrib — APPROVED** [review]

File: docs/quickstart/setup_spark.rst (hunk 2 above)
Comment (dolfinus) [reply docs/quickstart/setup_spark.rst]: Updated a note

File: README.rst (hunk 1 above)
Comment (TiGrib) [reply README.rst:45]: I mean mb better make a compare table in the README ?

Comment (dolfinus) [reply README.rst:45]: We'll probably add this comparison later

**Pull request merged**